proxy/pkg/util/util.go

```diff
@@ -506,19 +511,257 @@ func getUserClusterList(projectList []string) []string {
+
+	klog.Infof("Getting metrics access for user : %v", userName)
+
+	// get all metricsaccess ACLs for the user
+	// i.e every  metrics/<ns> on managedcluster CR defined for the user
+	// in the returned map -  key is managedcluster name , value is namespaces accesible on that cluster
+	metricsAccess, arerr := accessReviewer.GetMetricsAccess(token)
+	if arerr != nil {
+		klog.Errorf("Failed to get Metrics Access from Access Reviewer: %v", arerr)
+		return nil, arerr
+	}
+
+	klog.Infof("user <%v>  metrics-access: %v", userName, metricsAccess)
+
+	//if metrics access contains a key  "*" then the corresponding
+	// value i.e acls  apply to all managedclusters
+	if allClusterAcls, found := metricsAccess["*"]; found {
+		for mcName := range allManagedClusterNames {
+			if clusterAcls, ok := metricsAccess[mcName]; ok {
+				for _, allClusterAclItem := range allClusterAcls {
+					if !slices.Contains(clusterAcls, allClusterAclItem) {
+						clusterAcls = append(clusterAcls, allClusterAclItem)
+					}
+				}
+				metricsAccess[mcName] = clusterAcls
+			} else {
+				metricsAccess[mcName] = allClusterAcls
+			}
+		}
+		delete(metricsAccess, "*")
+	}
+
+	klog.Infof("user <%v>  metrics-access after processing for AllCluster acls: %v", userName, metricsAccess)
+
+	// for backward compatibility, support the existing access control mechanism
+	// i.e access to managedcluster project\namespace means access to all namespaces on that managedcluster
+
+	//get all managedcluster project/namespace user has access to
+	projectList, ok := GetUserProjectList(token)
+	klog.V(1).Infof("projectList from local mem cache = %v, ok = %v", projectList, ok)
+	if !ok {
+		projectList = FetchUserProjectList(token, reqUrl)
+		up := NewUserProject(userName, token, projectList)
+		UpdateUserProject(up)
+		klog.V(1).Infof("projectList from api server = %v", projectList)
+	}
+	klog.V(1).Infof("cluster list: %v", allManagedClusterNames)
+	klog.V(1).Infof("user <%s> project list: %v", userName, projectList)
+
+	clusterList := getUserClusterList(projectList)
+	klog.Infof("user <%v> have access to these clusters: %v", userName, clusterList)
+
+	// combine the  user project access list to the metrics access list
+	// project access by default give access to all namespaces ( i.e * ) unless
+	// scoped down to specific namespaces through metrics/<ns>  acl definition
+	for _, cluster := range clusterList {
+		if _, found := metricsAccess[cluster]; found {
+			continue
+		}
+
+		// metricAcls not found for this cluster,
+		// so add * for backward compatibility
+		// as user has access to the managedcluster project
+		metricsAccess[cluster] = []string{"*"}
+	}
+	klog.Infof("user <%v>  metrics-access after merging with cluster access list %v", userName, metricsAccess)
+
+	return metricsAccess, nil
+}
+
+// canAccessAllClusters check user have permission to access all clusters
+func canAccessAll(clusterNamespaces map[string][]string) bool {
```

Comment:
Function name in the comment doesn't match the actual function name.

Reply:
Fixed